detection_rules/utils.py

```diff
@@ -190,6 +190,10 @@ def cached(f):
 
     @functools.wraps(f)
     def wrapped(*args, **kwargs):
+        bypass_cache = kwargs.pop('bypass_cache', None)
```

Comment:
i think we talked about this once before,  but i'm not a fan.
it screws up with argument checking in IDEs and if you're conditionally caching and uncaching it's a weird sign.
either the function should be structured differently, or you just clear the entire cache by calling `myfunc.clear()` instead

Reply:
actually argument checking was working perfect for me in pycharm with this. But yes it is injecting an argument. Luckily I didn't end up using it so I can remove